### PR TITLE
Dependency updates

### PR DESCRIPTION
**Pull request by rosstimothy**
This PR replaces the following PRs opened by dependabot:

- #19310
- #19309
- #19308
- #19307
- #19306
- #19305
- #19304
- #19303
- #19302
- #19301
- #19300
- #19299
- #19298

### CODE REVIEW

**Comment by tigrato** [comment]
@rosstimothy I pushed a commit to force `wsStreamClient` to satisfy the change introduced by `https://github.com/gravitational/teleport/pull/19309` to `remotecommand.Executor`.

File: go.mod [hunk 1]
```diff
@@ -323,6 +324,7 @@ require (
 	github.com/russross/blackfriday/v2 v2.1.0 // indirect
 	github.com/ryszard/goskiplist v0.0.0-20150312221310-2dfbae5fcf46 // indirect
 	github.com/shabbyrobe/gocovmerge v0.0.0-20190829150210-3e036491d500 // indirect
+	github.com/siddontang/go v0.0.0-20180604090527-bdc77568d726 // indirect
```
[the remark this answers is not in the document]
Comment (codingllama) [reply go.mod:327]: It's slightly different: `siddontang/go/log` vs `siddontang/go-log/log`. How the packages are different or why are 2 of them I can't say. Nor why `go mod` keeps thinking this one is indirect.

[the remark this answers is not in the document]
Comment (codingllama) [reply go.mod:327]: Many thanks.

File: api/go.mod [hunk 2]
```diff
@@ -20,7 +20,7 @@ require (
 	go.opentelemetry.io/otel/trace v1.11.2
 	go.opentelemetry.io/proto/otlp v0.19.0
 	golang.org/x/exp v0.0.0-20221126150942-6ab00d035af9
-	golang.org/x/net v0.2.0
+	golang.org/x/net v0.3.1-0.20221206200815-1e63c2f08a10
```
Comment (codingllama) [inline api/go.mod:23]: I think we are missing a few direct deps here, sorry for not being more specific:

* go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp
* google.golang.org/genproto

**Review by codingllama — APPROVED** [review]
Thanks for the housekeeping!

**Review by codingllama — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Comment by codingllama** [comment]
Don't forget to tidy api/, if you haven't already.

File: go.mod [hunk 3]
```diff
@@ -142,20 +140,27 @@ require (
...
 	sigs.k8s.io/controller-runtime v0.13.1
 	sigs.k8s.io/controller-tools v0.10.0
 	sigs.k8s.io/yaml v1.3.0
 )
 
+// DO NOT UPDATE genproto without first updating firestore to 1.9.0
+// and fixing the deprecation of the Batch api in firestore 1.7.0.
+require (
+	cloud.google.com/go/firestore v1.6.1
+	google.golang.org/genproto v0.0.0-20221201164419-0e50fba7f41c
```
Comment (codingllama) [inline go.mod:161]: Add a dependabot ignore for genproto?

**Comment by codingllama** [comment]
> I've reverted the k8s updates as they produced linter issues:
> 
> ```
> level=warning msg="[runner] Can't run linter goanalysis_metalinter: buildir: failed to load package cache: could not load export data: no export data for \"sigs.k8s.io/controller-runtime/pkg/cache\""
> level=error msg="Running error: 1 error occurred:\n\t* can't run linter goanalysis_metalinter: buildir: failed to load package cache: could not load export data: no export data for \"sigs.k8s.io/controller-runtime/pkg/cache\"\n\n"```
> ```

No issues on my part, I'd rather get the rest of the updates going while we figure this out. Do you have an idea on what causes the problem?

**Pull request merged**